File: LeetCode/0551/main.c
#include<stdio.h>
#include<string.h>
#include<stdlib.h>
#include<stdbool.h>

bool checkRecord(char * s){
    int i;
    int cntA = 0, cntL = 0;
    for (i=0;i<strlen(s);i++){
        switch (*(s+i))
        {
        case 'A':
            if (++cntA >= 2) return false;
            cntL = 0;
            break;
        case 'L':
            if (++cntL >= 3) return false;
            break;
        case 'F':
            cntL = 0;
            break;
        default:
            cntL = 0;
            break;
        }
    }
    return true;
}

void main(void) {
    char *s = "PPALLPLA";
    printf("%d\n", checkRecord(s));
}

File: LeetCode/0551/main.go
package main

import "fmt"

func checkRecord(s string) bool {
	cntA, cntL := 0, 0
	for _, c := range s {
		switch c {
		case 'A':
			cntA++
			if cntA >= 2 {
				return false
			}
			cntL = 0
		case 'P':
			cntL = 0
		case 'L':
			cntL++
			if cntL >= 3 {
				return false
			}
		default:
			cntL = 0
		}
	}
	return true
}

func main() {
	fmt.Println(
		checkRecord("PPALLP"),
	)
}
